sentence_generator.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/python
+
+import re
+import random
+import sys
+
+# These mappings can get fairly large -- they're stored globally to
+# save copying time.
+
+# (tuple of words) -> {dict: word -> number of times the word appears following the tuple}
+# Example entry:
+#    ('eyes', 'turned') => {'to': 2.0, 'from': 1.0}
+# Used briefly while first constructing the normalized mapping
+tempMapping = {}
+
+# (tuple of words) -> {dict: word -> *normalized* number of times the word appears following the tuple}
+# Example entry:
+#    ('eyes', 'turned') => {'to': 0.66666666, 'from': 0.33333333}
+mapping = {}
+
+# Contains the set of words that can start sentences
+starts = []
+
+# We want to be able to compare words independent of their capitalization.
+def fixCaps(word):
+    # Ex: "FOO" -> "foo"
+    if word.isupper() and word != "I":
+        word = word.lower()
+        # Ex: "LaTeX" => "Latex"
+    elif word [0].isupper():
+        word = word.lower().capitalize()
+        # Ex: "wOOt" -> "woot"
+    else:
+        word = word.lower()
+    return word
+
+# Tuples can be hashed; lists can't.  We need hashable values for dict keys.
+# This looks like a hack (and it is, a little) but in practice it doesn't
+# affect processing time too negatively.
+def toHashKey(lst):
+    return tuple(lst)
+
+# Returns the contents of the file, split into a list of words and
+# (some) punctuation.
+def wordlist(words):
+    # f = open(filename, 'r')
```

Comment:
Often it's better to open and close files as-needed, avoiding this kind of commenting /uncommenting while developing.